### PR TITLE
add auto domain to infra PT.1

### PR DESCRIPTION
**Pull request by DaMandal0rian**
### **User description**
The PR adds auto domain to the terraform infra. The changes will be applied to devnet first.

___

### **PR Type**
enhancement

___

### **Description**
- Added support for an 'auto' domain alongside the existing 'evm' domain in various Terraform and shell script configurations.
- Updated environment variable and Docker compose setups to differentiate and handle multiple domain configurations.
- Enhanced domain node provisioner scripts to support the new domain setup.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Configuration changes
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>variables.tf</strong><dd><code>Extend domain configuration in variables</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

resources/devnet/variables.tf
<li>Added additional domain ID and label to support 'auto' domain <br>alongside 'evm'.<br>

</details>
    

  </td>
  <td><a href="https://github.com/subspace/infra/pull/311/files#diff-ae828b66c19dd29479737a7a40937ac48a79fdd36e4c3695b65a884a6c129692">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>bootstrap_node_evm_provisioner.tf</strong><dd><code>Update bootstrap node provisioner for multiple domains</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

templates/terraform/hetzner/bootstrap_node_evm_provisioner.tf
<li>Updated environment variable setup to differentiate between 'EVM' and <br>'AUTO' domain configurations.<br>

</details>
    

  </td>
  <td><a href="https://github.com/subspace/infra/pull/311/files#diff-fc6db2f194f6b9987d12a71ca29a199035f8e7003588cc724989293da84f52f0">+4/-4</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>domain_node_provisioner.tf</strong><dd><code>Enhance domain node provisioner for multi-domain support</code>&nbsp; </dd></summary>
<hr>

templates/terraform/hetzner/domain_node_provisioner.tf
- Modified domain node provisioner to handle multiple domains.

</details>
    

  </td>
  <td><a href="https://github.com/subspace/infra/pull/311/files#diff-5706b6156e87e7ce6632980202e39ed0637b1d2afb0cf1253b0cd76fffa2e44c">+4/-5</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>bootstrap_node_evm_provisioner.tf</strong><dd><code>Refactor bootstrap node provisioner for domain clarity</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

templates/terraform/network-primitives/bootstrap_node_evm_provisioner.tf
<li>Adjusted domain-related environment variables for clarity and <br>separation.<br>

</details>
    

  </td>
  <td><a href="https://github.com/subspace/infra/pull/311/files#diff-faa8d3100407bc4203aee1847804d330c0c7942c87ccde5151250146c2986f95">+4/-4</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>domain_node_provisioner.tf</strong><dd><code>Update domain node provisioner for multiple domains</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

templates/terraform/network-primitives/domain_node_provisioner.tf
<li>Updated domain node provisioner scripts to support multiple domain IDs <br>and labels.<br>

</details>
    

  </td>
  <td><a href="https://github.com/subspace/infra/pull/311/files#diff-d23fe9738ab570ba931c77c031cf37c88bb459137837ce1cea86e8a564e324c1">+4/-4</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>create_bootstrap_node_evm_compose_file.sh</strong><dd><code>Add 'auto' domain support in Docker compose script</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

templates/scripts/create_bootstrap_node_evm_compose_file.sh
- Added support for 'auto' domain in Docker compose configuration.

</details>
    

  </td>
  <td><a href="https://github.com/subspace/infra/pull/311/files#diff-579c50457274d21a7b133283ca6ad7754890686962990edfeb597b93b725b762">+14/-2</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>create_domain_node_compose_file.sh</strong><dd><code>Extend Docker compose configuration for multiple domains</code>&nbsp; </dd></summary>
<hr>

templates/scripts/create_domain_node_compose_file.sh
- Extended Docker compose configuration to handle multiple domains.

</details>
    

  </td>
  <td><a href="https://github.com/subspace/infra/pull/311/files#diff-57a235dd3c69f6e6cffe833934963a7f6d5e2e5115dcd1ee0afffe95f3f6ec6a">+20/-2</a>&nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: templates/scripts/create_bootstrap_node_evm_compose_file.sh [hunk 1]
```diff
@@ -192,7 +192,19 @@ if [ "${enable_domains}" == "true" ]; then
       echo "      \"--reserved-nodes\", \"${addr}\"," >> ~/subspace/docker-compose.yml
       echo "      \"--bootstrap-nodes\", \"${addr}\"," >> ~/subspace/docker-compose.yml
     done
-
+    # auto domain
+      echo '      "--",'
```
Comment (vedhavyas) [inline templates/scripts/create_bootstrap_node_evm_compose_file.sh]: This is not correct unfotunately

We would need to create two compose file. One for EVM and one for AutoID
Our node currently does not support running multiple domains in the single binary

File: resources/devnet/main.tf [hunk 2]
```diff
@@ -37,6 +37,23 @@ module "devnet" {
     disk-volume-type   = var.disk_volume_type
   }
 
+  bootstrap-node-autoid-config = {
```
Comment (vedhavyas) [inline resources/devnet/main.tf:40]: I was actually hoping we can merge the nova and autoID in to a single type. There is not much difference except domain-id

File: templates/scripts/create_autoid_node_compose_file.sh [hunk 3]
```diff
@@ -0,0 +1,176 @@
+#!/bin/bash
```
Comment (vedhavyas) [inline templates/scripts/create_autoid_node_compose_file.sh:1]: Why not use the EVM script ?
There is no difference in them

**Review by vedhavyas — COMMENTED** [review]
This PR changes can be much smaller if we merged the nova and autoID config. This will pave way for much more composable infra for future domains too

File: resources/devnet/main.tf (hunk 2 above)
Comment (DaMandal0rian) [reply resources/devnet/main.tf:40]: If we are running both concurrently how will it work?

> This is not correct unfotunately

We would need to create two compose file. One for EVM and one for AutoID
Our node currently does not support running multiple domains in the single binary

**Pull request merged**